tests/test_core.py

```diff
@@ -5,17 +5,17 @@
 import shutil
 import tempfile
 import unittest
```

Comment:
Since this PR is aiming to switch from unittest to pytest, I'd look at the uses of `unittest` within this module and see if you can replace them with pytest equivalents.

One example I see is the usage of the `@unittest.skipUnless` decorator - this can be replaced by "inverting" the logic on the test case and using `@pytest.mark.skipif` instead.